invenio_app_rdm/utils/checks.py

```diff
@@ -0,0 +1,82 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2025 CERN.
+#
+# Invenio-Checks is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+"""Checks utilities."""
+
+from flask import current_app, g
+from invenio_communities.proxies import current_communities
+from invenio_rdm_records.requests import CommunityInclusion, CommunitySubmission
+from sqlalchemy import case
+
+
+def resolve_checks(record_uuid, request, community=None):
+    """Resolve the checks for this draft/record related to the community and the request."""
+    # Early exit if checks are not enabled.
+    enabled = current_app.config.get("CHECKS_ENABLED", False)
+
+    if not enabled:
+        return None
+
+    # Early exit if not draft submission nor record inclusion
+    request_type = request["type"]
+    is_draft_submission = request_type == CommunitySubmission.type_id
+    is_record_inclusion = request_type == CommunityInclusion.type_id
+
+    if not is_draft_submission and not is_record_inclusion:
+        return None
+
+    # Early exit if there is no record UUID (for instance for some closed requests)
+    if not record_uuid:
+        return None
+
+    # Resolve the target community from the request if the community was not passed as an argument
+    if not community:
+        community_uuid = request["receiver"]["community"]
+        community = current_communities.service.read(
+            id_=community_uuid, identity=g.identity
+        )
+
+    # Collect the community UUID and the potential parent community UUID
+    communities = []
+    community_parent_id = community.to_dict().get("parent", {}).get("id")
+    if community_parent_id:
+        # Add the parent community first for later ordering of check configs
+        communities.append(community_parent_id)
+    communities.append(community.id)
+
+    # Early exit if no check config found for the communities
+    from invenio_checks.models import CheckConfig, CheckRun
```

Comment:
why local import is needed?

Reply:
`invenio-checks` is not a dependency of `invenio-app-rdm` by default, so this is only accessed if you have set `CHECKS_ENABLED = True`. We can add the dependency now, since we're getting more production usage.